### PR TITLE
@kanaabe => Panel video

### PR DESCRIPTION
**Pull request by eessex**
Adds support for videos in panel units.  An optional coverImage is displayed if available, the video plays when hovering on the unit.

TODO:
- Tracking event for video play
- Update Positron to allow cover images in video panels

panel-video

### CODE REVIEW

File: src/Components/Publishing/Display/__test__/DisplayPanel.test.tsx [hunk 1]
```diff
@@ -1,10 +1,15 @@
...
-it("renders the display panel", () => {
+it("renders the display panel with an image", () => {
   const displayPanel = renderer.create(<DisplayPanel unit={UnitPanel} campaign={Campaign} />).toJSON()
   expect(displayPanel).toMatchSnapshot()
 })
+
+it("renders the display panel with video", () => {
+  const displayPanel = renderer.create(<DisplayPanel unit={UnitPanelVideo} campaign={Campaign} />).toJSON()
+  expect(displayPanel).toMatchSnapshot()
+})
```
Comment (kanaabe) [inline src/Components/Publishing/Display/__test__/DisplayPanel.test.tsx]: Might be helpful to start integrating some unit tests along with snapshot tests, especially for interactive components.

File: src/Components/Publishing/Display/DisplayPanel.tsx [hunk 2]
```diff
@@ -12,12 +12,17 @@ interface DisplayPanelProps extends React.HTMLProps<HTMLDivElement> {
 }
 
 interface DivUrlProps extends React.HTMLProps<HTMLDivElement> {
-  imageUrl: string
-  hoverImageUrl: string
+  imageUrl?: string
+  hoverImageUrl?: string
+  coverUrl?: string
+  onMouseEnter: any
+  onMouseLeave: any
```
Comment (kanaabe) [inline src/Components/Publishing/Display/DisplayPanel.tsx]: How does one play the video on mobile?

Comment (eessex) [reply src/Components/Publishing/Display/DisplayPanel.tsx]: Oh good point, cc @owendodd

File: src/Components/Publishing/Display/DisplayPanel.tsx [hunk 3]
```diff
@@ -132,3 +186,15 @@ const Body = styled.div`
...
+const VideoContainer = Image.extend`
+position: relative;
+overflow: hidden;
+display: flex;
+align-items: center;
+video {
+  object-fit: cover;
+  object-position: 50%;
+  width: 100%;
+}
```
Comment (kanaabe) [inline src/Components/Publishing/Display/DisplayPanel.tsx]: Indentation looks off here

File: src/Components/Publishing/Display/DisplayPanel.tsx (hunk 2 above)
Comment (eessex) [reply src/Components/Publishing/Display/DisplayPanel.tsx]: There is also no mobile alternative for the image layout, assuming that is ok

File: src/Components/Publishing/Display/__test__/DisplayPanel.test.tsx (hunk 1 above)
Comment (eessex) [reply src/Components/Publishing/Display/__test__/DisplayPanel.test.tsx]: Yeah was looking at testing for this, but had trouble getting access to the video ref -- if you're up to pair today I'd be glad to have help figuring this out.

**Comment by kanaabe** [comment]
Looks like you got some help from @orta on the test - looking forward to seeing that! 

I guess the remaining question now is the mobile video play. Feel free to address that in a separate PR along with the tracking if you want. Feel free to merge after the test + ❌  is fixed!

**Comment by eessex** [comment]
Actually didn't get the tests working -- the @track() is preventing jest mocks from acting as expected. Have been conversation with @orta and @damassi about this issue. Going to leave unit tests for a separate PR, as they are blocking atm.

File: src/Components/Publishing/Display/DisplayPanel.tsx (hunk 2 above)
Comment (eessex) [reply src/Components/Publishing/Display/DisplayPanel.tsx]: A quick solution could be adding a play icon
screen shot 2017-10-31 at 9 06 16 pm

Comment (owendodd) [reply src/Components/Publishing/Display/DisplayPanel.tsx]: Yeah good call. This would work well!

**Pull request merged**